File: backend/express/api/actions.js
import db from '../db/connect.js'

export const getCategories = async (req, res) => {
    try {
        const [result] = await db.promise().query('select * from kategorie');
        res.status(200).json(result);
    } catch (err) {
        console.error("błąd przy pobieraniu kategorii", err);
        res.status(500).json({error: "bład bazy przy pobieraniu kategorii"});
    }
}

export const getUsers = async (req, res) => {
    try {
        const [result] = await db.promise().query('select * from uzytkownik');
        res.status(200).json(result);
    } catch (err) {
        console.error("błąd przy pobieraniu użytkownikow", err);
        res.status(500).json({error: "bład bazy przy pobieraniu uzytkownikow"});
    }
}

export const getAds = async (req, res) => {
    try {
        const [result] = await db.promise().query(`
            select ogloszenie.id, ogloszenie.tytul, ogloszenie.tresc, uzytkownik.telefon, ogloszenie.kategoria
            from ogloszenie
            join uzytkownik on ogloszenie.uzytkownik_id = uzytkownik.id
            `);
        res.status(200).json(result);
    } catch (err){
        console.error("błąd przy pobieraniu ogloszen", err);
        res.status(500).json({error: "bład bazy przy pobieraniu ogloszen"});
    }
}

export const addAd = async (req, res) => {
    try {

    } catch (err) {
        console.error("błąd przy dodwaniu ogloszenia", err);
        res.status(500).json({error: "bład bazy przy dodwaniu ogloszenia"});
    }
}

File: backend/flask/api/actions.py
from flask import jsonify, request
from db.connect import connect
import pymysql

def get_categories():
    conn = None
    try:
        conn = connect()
        with conn.cursor() as cursor:
            cursor.execute("SELECT * FROM kategorie")
            results = cursor.fetchall()
            return jsonify(results), 200
    except Exception as e:
        print(f"błąd przy pobieraniu kategorii: {e}")
        return jsonify({"error": str(e)}), 500
    finally:
        if conn: conn.close()

def get_users():
    conn = None
    try:
        conn = connect()
        with conn.cursor() as cursor:
            cursor.execute("SELECT * FROM uzytkownik")
            results = cursor.fetchall()
            return jsonify(results), 200
    except Exception as e:
        print(f"błąd przy pobieraniu użytkownikow: {e}")
        return jsonify({"error": str(e)}), 500
    finally:
        if conn: conn.close()

def get_ads():
    conn = None
    try:
        conn = connect()
        
        sql = """
            SELECT ogloszenie.id, ogloszenie.tytul, ogloszenie.tresc, 
                   uzytkownik.telefon, ogloszenie.kategoria
            FROM ogloszenie 
            JOIN uzytkownik ON ogloszenie.uzytkownik_id = uzytkownik.id
        """
        with conn.cursor() as cursor:
            cursor.execute(sql)
            results = cursor.fetchall()
            return jsonify(results), 200

    except Exception as e:
        print(f"błąd przy pobieraniu ogloszen: {e}")
        return jsonify({"error": str(e)}), 500
    finally:
        if conn: conn.close()

def add_ad():
    conn = None
    try:
        data = request.json
        
        conn = get_connection()
        sql = """
            INSERT INTO ogloszenie (uzytkownik_id, kategoria, tytul, tresc) 
            VALUES (%s, %s, %s, %s)
        """
        values = (data['uzytkownik_id'], data['kategoria'], data['tytul'], data.get('tresc', ''))
        
        with conn.cursor() as cursor:
            cursor.execute(sql, values)
            conn.commit()
            new_id = cursor.lastrowid
            
        return jsonify({'message': 'Ad added', 'id': new_id}), 201

    except Exception as e:
        print(f"błąd przy dodwaniu ogloszenia: {e}")
        return jsonify({"error": str(e)}), 500
    finally:
        if conn: conn.close()